Pellicola/Classes/ImagesDataStorage.swift

```diff
@@ -80,15 +81,13 @@ class ImagesDataStorage: NSObject {
             DispatchQueue.main.async {
                 block(urls)
             }
+            self.clear()
         }
     }
     
     func clear() {
         images.removeAll()
-        
-        fileQueue.async { [weak self] in
-            guard let self = self else { return }
-            self.fileHandler.deleteAllImages()
-        }
+        fileActions.forEach { $0.cancel() }
+        fileActions.removeAll()
```

Comment:
⚠️ If you want to cancel a `DispatchWorkItem` you need to manually call `cancel` on it and check within its block for the value of its `isCancelled`.